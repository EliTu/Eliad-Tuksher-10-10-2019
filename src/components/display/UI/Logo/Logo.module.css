@import '../../../../variables.css';

.LogoStyles {
    margin: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    justify-items: center;
}

.LogoStyles div {
    margin-right: 1rem;
    width: 4.5rem;
    height: 4.5rem;
}

.LogoStyles p:nth-child(1) {
    width: 100%;
    font-size: var(--text-size-extra-large);
}

.LogoStyles p:nth-child(2) {
    font-size: var(--text-size-large);
}