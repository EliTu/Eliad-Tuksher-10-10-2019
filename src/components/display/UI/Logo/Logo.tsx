import React from 'react';
import Icon from '../Icon/Icon';
import { faRainbow } from '@fortawesome/free-solid-svg-icons';
import styles from './Logo.module.css';

const Logo: React.FC = () => {
	const { LogoStyles } = styles;

	return (
		<div className={LogoStyles}>
			<Icon iconType={faRainbow} size={'4x'} />
			<ul>
				<li>Herolo's</li>
				<li>Weather app</li>
			</ul>
		</div>
	);
};

export default Logo;
