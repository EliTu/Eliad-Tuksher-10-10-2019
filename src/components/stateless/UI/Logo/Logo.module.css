.LogoStyles {
    margin: 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    justify-items: center;
}

.testDiv {
    margin-right: 1rem;
    width: 4.5rem;
    height: 4.5rem;
    border: 1px solid black;
}