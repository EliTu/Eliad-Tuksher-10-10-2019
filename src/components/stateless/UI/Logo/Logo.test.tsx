import React from 'react';
import { shallow } from 'enzyme';
import Logo from './Logo';

describe('Logo component', () => {
	let component;
	beforeEach(() => {
		component = shallow(<Logo />);
	});

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
		expect(component).toHaveClassName('LogoStyles');
	});

	it('should render one span tag with a text', () => {
		const span = component.children().find('span');

		expect(span.length).toBe(1);
		expect(span.text()).toBe('Herolo Weather app!');
	});

	it('should render one img tag, containing a logo (currently a div for testing)', () => {
		const div = component.children().find('div');

		expect(div.length).toBe(1);
		expect(div).toHaveClassName('testDiv');
	});
});
