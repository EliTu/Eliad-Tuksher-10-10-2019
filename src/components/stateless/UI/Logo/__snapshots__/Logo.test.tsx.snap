// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render without errors 1`] = `
<div
  className="LogoStyles"
>
  <div
    className="testDiv"
  />
  <span>
    Herolo Weather app!
  </span>
</div>
`;
